### ISSUE TITLE
[vim9script] cannot call class member of funcref type directly, E1325: Method not found on class

### ISSUE THREAD

**Issue by qeatzy** [vim/vim#12324]
### Steps to reproduce

1. create a file with any name. eg, `repro.vim` with below contents.

```
vim9script

if !exists('g:fns')
    g:fns = [['main.go', 'main_windows.go', 'version.go', 'conn/bind_std.go', 'conn/bind_windows.go', 'conn/boundif_android.go', 'conn/conn.go', 'conn/controlfns.go', 'conn/controlfns_linux.go', 'conn/controlfns_unix.go', 'conn/controlfns_windows.go', 'conn/default.go'], 0]
endif

def Open(filename: string)
    :e `=filename`
enddef

export class FloatWindow
    public this.fns: list<any>
    this.idx: number
    this.cb: func(any) = Open

    # idx = step, return fns[new idx]
    def To(step: number, do: bool = 0): any
        var x = step % len(this.fns)
        x = x >= 0 ? x : x + len(this.fns) 
        var ret = this.fns[this.idx]
        this.idx = x
        if do
            # this.cb(ret)
            # error E1325: Method not found on class "FloatWindow": cb(ret)
            # below is alternative
            var F = this.cb
            F(ret)
        endif
        return ret
    enddef

    # idx += step, return fns[old idx]
    def Inc(step: number): any
        var x = (this.idx + step) % len(this.fns)
        x = x >= 0 ? x : x + len(this.fns) 
        var ret = this.fns[this.idx]
        this.idx = x
        return ret
    enddef

endclass

var fw = FloatWindow.new(g:fns[0])
# echo fw
fw.To(3, 1)
echo mapnew(fw.fns, 'fw.Inc(1)')
g:fw = fw

```

2 `vim --clean -c "so repro.vim"`

### Expected behaviour

Inside method `def To(step: number, do: bool = 0): any`

below is the original version throws error, also try change this.cb to this.Cb, same result
```
        if do
            # E1325: Method not found on class "FloatWindow": cb(ret)
            this.cb(ret)
        endif
```
below is workaround to suppress the `E1325: Method not found on class` message
```
        if do
            # workaround
            var F = this.cb
            F(ret)
        endif
```

Here is the question, is this behavior -- not able to call class member of Funcref type directly -- final and cannot be fixed, or it's just due to some imperfection/oblivion/difficulty in current implementation?

### Version of Vim

[9.0.1488_x86] VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Apr 25 2023 23:12:17) MS-Windows 32-bit GUI version with OLE support Included patches: 1-1488

### Environment

Not matter

**Comment by yegappan** [comment]
I am not able to reproduce this issue with the latest version of Vim.  Can you try using the latest version of Vim?

**Comment by errael** [comment]
The example has the workaround. To reproduce the problem, un-comment `this.cb(ret)` on line 23, and comment out the alternative on lines 26-27.

**Comment by chrisbra** [comment]
closed by 29bb67f1beefc7fd393dbfd9ee77d92f1db3a3c0
